### ISSUE TITLE
[Question] What is the __check_nan method in the env_checker doing?

### ISSUE THREAD

**Issue by FieteO** [DLR-RM/stable-baselines3#1402]
### ❓ Question

In the `env_checker`, there is the an invokation to [check_nan(env)](https://github.com/DLR-RM/stable-baselines3/blob/c5adad82b2c1733bd8add7e2eeb49a895b635856/stable_baselines3/common/env_checker.py#L387)
```python
    try:
        check_for_nested_spaces(env.observation_space)
        # The check doesn't support nested observations/dict actions
        # A warning about it has already been emitted
        _check_nan(env)
    except NotImplementedError:
        pass
```
This is then effectively invoking the `env.step` method ten times:
```python
def _check_nan(env: gym.Env) -> None:
    """Check for Inf and NaN using the VecWrapper."""
    vec_env = VecCheckNan(DummyVecEnv([lambda: env]))
    for _ in range(10):
        action = np.array([env.action_space.sample()])
        _, _, _, _ = vec_env.step(action)
```
What's the purpose of this? I don't see any assertions in there, so is it just assuring that no exception is thrown?

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is minimal and working
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
> What's the purpose of this? I don't see any assertions in there, so is it just assuring that no exception is thrown?

The answer is in the `VecCheckNan` (it is explained in the doc the purpose of that class).

**Comment by FieteO** [comment]
Thanks, so this is checking if the environment could be run in a vectorized context, even though that may not be intended at all, right?

**Comment by araffin** [comment]
> Ah I see. What's the reasoning behind doing that for 10 steps? Is that number arbitrary?

yes arbitrary, bigger than one (to check for errors that happens after some steps) but small enough that it is fast to run.
